src/main/java/io/jmix/uisamples/view/entity/food/FoodDetailView.java

```diff
@@ -0,0 +1,64 @@
+package io.jmix.uisamples.view.entity.food;
+
+
+import com.vaadin.flow.component.avatar.Avatar;
+import com.vaadin.flow.component.html.H2;
+import com.vaadin.flow.server.StreamResource;
+import io.jmix.flowui.component.SupportsTypedValue;
+import io.jmix.flowui.component.textfield.TypedTextField;
+import io.jmix.flowui.component.upload.FileUploadField;
+import io.jmix.flowui.kit.component.upload.event.FileUploadSucceededEvent;
+import io.jmix.flowui.view.*;
+import io.jmix.uisamples.entity.Food;
+
+import org.apache.commons.lang3.StringUtils;
+
+import java.io.ByteArrayInputStream;
+import java.util.Objects;
+
+@DialogMode(minWidth = "15em", minHeight = "10pm")
+@EditedEntityContainer("foodDc")
+@ViewController("Food.detail")
+@ViewDescriptor("food-detail-view.xml")
+public class FoodDetailView extends StandardDetailView<Food> {
+
+    @ViewComponent
+    private Avatar foodIcon;
+    @ViewComponent
+    private FileUploadField foodIconUpload;
+    @ViewComponent
+    private H2 foodTitle;
+
+    @Subscribe
+    public void obBeforeShow(final BeforeShowEvent event) {
+        initIconFields();
+    }
+
+    @Subscribe("titleField")
+    public void onNameFieldTypedValueChange(final SupportsTypedValue.TypedValueChangeEvent<TypedTextField<String>, String> event) {
+        foodTitle.setText(event.getValue());
+    }
+
+    @Subscribe("foodIconUpload")
+    public void onFoodIconUploadFileUploadSucceeded(final FileUploadSucceededEvent<FileUploadField> event) {
```

Comment:
Date binding has left the chat...


Replace with: 
```xml
<fileUploadField id="foodIconUpload" dataContainer="foodDc" property="icon" uploadIcon="UPLOAD" uploadText=""/>
```

After that....
`ItemPropertyChangeEvent` would be much better:
```java
    @Subscribe(id = "foodDc", target = Target.DATA_CONTAINER)
    protected void onPropertyChangeListener(InstanceContainer.ItemPropertyChangeEvent<Food> event) {
        if ("icon".equals(event.getProperty())) {
            initIconFields();
        }
    }
```